### PR TITLE
Add integration test for trashbin

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of owncloud/core#27069
* redo of #3893

### CODE REVIEW

File: build/integration/features/favorites.feature [hunk 1]
```diff
@@ -134,3 +134,16 @@ Feature: favorite
...
+        And As an "admin"
+        And user "user0" exists
+        And user "user1" exists
+        And user "user0" created a folder "/shared"
+        And User "user0" moved file "/textfile0.txt" to "/shared/shared_file.txt"
+        And folder "/shared" of user "user0" is shared with user "user1"
+        And user "user1" favorites element "/shared/shared_file.txt"
+        When User "user1" moved file "/shared/shared_file.txt" to "/taken_out.txt"
+        Then user "user1" in folder "/" should have favorited the following elements
+            | /taken_out.txt |
```
Comment (MorrisJobke) [inline build/integration/features/favorites.feature]: This tests seems to fail. The path is prefixed with `/remote.php/webdav`. @icewind1991 @rullzer Is this correct?

Comment (rullzer) [inline build/integration/features/favorites.feature]: Yes that is the correct webdav path

Comment (MorrisJobke) [inline build/integration/features/favorites.feature]: @rullzer I adjusted the path in the tests.

**Comment by ChristophWurst** [comment]
```
--- Failed scenarios:
/drone/src/github.com/nextcloud/server/build/integration/features/favorites.feature:137
```

**Comment by MorrisJobke** [comment]
Aaah ... now it says:

```
/remote.php/webdav/remote.php/webdav/taken_out.txt is not in report answer
```

before prepending the `/remote.php/webdav` it said:

```
/taken_out.txt is not in report answer
```
🙈

**Comment by MorrisJobke** [comment]
Let's see if it works now

**Review by rullzer — APPROVED** [review]
Lets do this!

File: build/integration/features/favorites.feature (hunk 1 above)
Comment (MorrisJobke) [inline build/integration/features/favorites.feature]: This one is forbidden. I will move this to a separate PR to maybe debug this.

**Comment by MorrisJobke** [comment]
I will also rebase to get #4302 resolved.

**Comment by MorrisJobke** [comment]
:+1: from me as well for this

**Pull request merged**